src/App.js

```diff
@@ -1,64 +1,42 @@
-import React, { Component } from 'react';
-import PropTypes from 'prop-types';
-import './App.css';
-import HeaderForm from './components/HeaderForm';
-import TubeMapContainer from './components/TubeMapContainer';
-import CustomizationAccordion from './components/CustomizationAccordion';
-import { dataOriginTypes } from './enums';
-import * as tubeMap from './util/tubemap';
-import config from './config.json';
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import isEqual from "react-fast-compare";
+
+import "./App.css";
+import HeaderForm from "./components/HeaderForm";
+import TubeMapContainer from "./components/TubeMapContainer";
+import { urlParamsToViewTarget } from "./components/CopyLink";
+import CustomizationAccordion from "./components/CustomizationAccordion";
+import { dataOriginTypes } from "./enums";
+import * as tubeMap from "./util/tubemap";
+import config from "./config.json";
+// Data path  - see server.js and config.json
 
 class App extends Component {
   constructor(props) {
     super(props);
-    const ds = config.DATA_SOURCES[0];
-    let xgFile = ds.xgFile;
-    let region = ds.defaultPosition;
-    let gamFile = undefined;
-    if(ds.gamFile){
-      gamFile = ds.gamFile;
-    }
-    let gbwtFile = undefined;
-    if(ds.gbwtFile){
-      gbwtFile = ds.gbwtFile;
-    }
-    let bedFile = undefined;
-    if(ds.bedFile){
-      bedFile = ds.bedFile;
-    }
-    let dataPath = 'default';
-    if(ds.useMountedPath){
-      dataPath = 'mounted';
-    }
+
+    const ds =
+      urlParamsToViewTarget(document.location) ?? config.DATA_SOURCES[0];
     this.state = {
       // These describe the files on the server side that we are working on.
-      fetchParams: {
-        // This is the query (like path:start-end) we are displaying.
-        region: region,
-        xgFile: xgFile,
-        gbwtFile: gbwtFile,
-        gamFile: gamFile,
-        bedFile: bedFile,
-        // This is the type of data paths we are working with, such as "mounted".
-        // All the paths are scoped to a type on the server side.
-        dataPath: dataPath
-      },
       // This is a little like dataPath, but lets us toggle between data from
       // the server and local test data. TODO: Unify?
```

Comment:
This should apply to `dataOrigin` and not `viewTarget`, right? Also, maybe now it needs to say what it means by `dataPath` (i.e. that it's part of `viewTarget` here?).